File: 122/solution.go
package _122

func maxProfit(prices []int) int {
	sell := 0
	buy := -prices[0]
	for i := 1; i < len(prices); i++ {
		tmp := buy
		buy = max(buy, sell-prices[i])
		sell = max(sell, tmp+prices[i])
	}
	return sell
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: 122/solution.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        sell = 0
        buy = -prices[0]
        for i in range(1, len(prices)):
            tmp = buy
            buy = max(buy, sell - prices[i])
            sell = max(sell, tmp + prices[i])
        return sell
